File: app/src/main/java/com/example/cartier/marveldc/adapters/PagerAdapter.java
package com.example.cartier.marveldc.adapters;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import com.example.cartier.marveldc.fragments.DCFragment;
import com.example.cartier.marveldc.fragments.MarvelFragment;


/**
 * Created by Cartier on 12/6/2016.
 */
public class PagerAdapter extends FragmentStatePagerAdapter {
    int mNumOfTabs;

    public PagerAdapter(FragmentManager fm, int NumOfTabs) {
        super(fm);
        this.mNumOfTabs = NumOfTabs;
    }

    @Override
    public Fragment getItem(int position) {

        switch (position) {
            case 0:
                MarvelFragment tab1 = new MarvelFragment();
                return tab1;
            case 1:
                DCFragment tab2 = new DCFragment();
                return tab2;
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return mNumOfTabs;
    }
}

File: app/src/main/java/com/example/cartier/marveldc/adapters/PagerAdapter.kt
package com.example.cartier.marveldc.adapters

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter

import com.example.cartier.marveldc.fragments.DCFragment
import com.example.cartier.marveldc.fragments.MarvelFragment


class PagerAdapter(fm: FragmentManager, internal var mNumOfTabs: Int) : FragmentStatePagerAdapter(fm) {

    override fun getItem(position: Int): Fragment? {

        when (position) {
            0 -> {
                val tab1 = MarvelFragment()
                return tab1
            }
            1 -> {
                val tab2 = DCFragment()
                return tab2
            }
            else -> return null
        }
    }

    override fun getCount(): Int {
        return mNumOfTabs
    }
}
